keras/callbacks.py

```diff
@@ -779,11 +796,35 @@ def on_epoch_end(self, epoch, logs=None):
                     self.writer.add_summary(summary_str, epoch)
                     i += self.batch_size
 
-        if self.embeddings_freq and self.embeddings_ckpt_path:
+        if self.embeddings_freq and self.embeddings_data is not None:
             if epoch % self.embeddings_freq == 0:
-                self.saver.save(self.sess,
-                                self.embeddings_ckpt_path,
-                                epoch)
+
+                embeddings_data = self.embeddings_data
+                n_samples = embeddings_data.shape[0]
+
+                i = 0
+                while i < n_samples:
+                    if type(self.model.input) == list:
+                        feed_dict = {model_input: self.embeddings_data
+                                     for model_input in list(self.model.input)}
+                    else:
+                        feed_dict = {self.model.input: self.embeddings_data}
+
+                    step = min(self.batch_size, n_samples - i)
+                    if self.model.uses_learning_phase:
+                        feed_dict[K.learning_phase()] = False
+                        # do not slice the learning phase
+                        batch_val = [x[i:i + step] for x in embeddings_data[:-1]]
+                        batch_val.append(embeddings_data[-1])
+                    else:
+                        batch_val = [x[i:i + step] for x in embeddings_data]
```

Comment:
`batch_val` is not used afterwards. I would have expected it to be passed to the `feed_dict`.

Reply:
Oops. Of course. Thanks for pointing this out. That explains why tests were still green after this change. 

This actually exposes a problem: The `tf.Variable` holding the `embedding` is instantiated with `shape = (n_samples, embedding_dim)`. When I pass the `feed_dict` with only `self.batch_size` samples, this blows up because dimensions (number of samples) are not matched. ☹️

I could not, yet, find a way to save the `tf.Variable` holding the `embedding` in batches.

Theoretically, we could just have a `tf.Variable` for every batch, compute the embedding for `self.batch_size` samples and concatenate them into the final embedding with `n_samples`.

There just doesn't seem to be a way using `tf.train.Saver`.

Any ideas?

Reply:
This probably won't help with your actual problem, but the following PR was very helpful when I needed to fix feed dict issues: https://github.com/fchollet/keras/pull/7064

Reply:
You should assign the embedding_input, which is a batch instead of all data to the appropriate slice of the embeddings variables. I.e. instead of ```tf.assign(embedding, embedding_input)```, you could write ```tf.assign(embeddings[batch_id:batch_id + step], embedding_input)```, with ```batch_id = tf.placeholder(tf.int32)```, that you feed the `i` to when actually iterating over batches. I've written a test with a minimal stand-alone example. Let me know if you want me to paste it here for clarification.

Reply:
Similarly, I could imagine that it might be possible to provide a slice to tf.train.Saver instead of the entire embedding variable.

Reply:
@chleibig, I'd appreciate if you could post your example (if you still have this lying around.)

Reply:
Unfortunately, I don't have this lying around anymore.